From 37b80e4399b3dfa857c9b51866c75ce8d375bb9c Mon Sep 17 00:00:00 2001
From: Stephan Linz <linz@li-pro.net>
Date: Sat, 16 Feb 2013 20:36:54 +0100
Subject: [PATCH] arm-oxnas: move default configuration

Move all the default predefinitions from boards
config.mk to boards config.h

Signed-off-by: Stephan Linz <linz@li-pro.net>
---
 board/oxnas/Makefile       |    4 +-
 board/oxnas/config.mk      |   28 +++-------------
 board/oxnas/ide-810.c      |    4 ++
 board/oxnas/platform-800.S |    4 ++
 board/oxnas/platform-810.S |    4 ++
 include/configs/oxnas.h    |   78 +++++++++++++++++++++++++++++++++++++++----
 6 files changed, 89 insertions(+), 33 deletions(-)
 mode change 100644 => 100755 mkconfig

diff --git a/board/oxnas/Makefile b/board/oxnas/Makefile
index 6f80690..0919348 100644
--- a/board/oxnas/Makefile
+++ b/board/oxnas/Makefile
@@ -29,8 +29,8 @@ include $(TOPDIR)/config.mk
 
 LIB	= lib$(BOARD).a
 
-OBJS	:= oxnas.o eth.o ide-$(NAS_VERSION).o
-SOBJS	:= platform-$(NAS_VERSION).o
+OBJS	:= oxnas.o eth.o ide-810.o
+SOBJS	:= platform-800.o platform-810.o
 
 $(LIB):	$(OBJS) $(SOBJS)
 	$(AR) crv $@ $^
diff --git a/board/oxnas/config.mk b/board/oxnas/config.mk
index b404b94..318c94d 100644
--- a/board/oxnas/config.mk
+++ b/board/oxnas/config.mk
@@ -1,25 +1,7 @@
-TEXT_BASE = 0x48d00000
-
-PLL400 ?= 733333333
-RPSCLK ?=  25000000
-
-NAS_VERSION ?= 810
-FPGA ?= 0
-FPGA_ARM_CLK ?= 25000000
-
-PROBE_MEM_SIZE ?= 1
-MEM_SIZE ?= 64	# Memory size in megabytes if probing is not enabled
-MEM_ODT ?= 150
+#
+# OXNAS boards (OXE800/OXE810)
+#
 
-USE_SATA ?= 1
-USE_SATA_ENV ?= 1
-USE_FLASH ?= 1
-
-LINUX_ROOT_RAIDED ?= 1
-
-USE_EXTERNAL_UART ?= 0
-INTERNAL_UART ?= 2
-
-TEST_BRD ?= 0	# Only significant for OX800
+TEXT_BASE = 0x48d00000
 
-PLATFORM_CPPFLAGS += -DLINUX_ROOT_RAIDED=$(LINUX_ROOT_RAIDED) -DMEM_ODT=$(MEM_ODT) -DPROBE_MEM_SIZE=$(PROBE_MEM_SIZE) -DNAS_VERSION=$(NAS_VERSION) -DFPGA=$(FPGA) -DFPGA_ARM_CLK=$(FPGA_ARM_CLK) -DINTERNAL_UART=$(INTERNAL_UART) -DUSE_EXTERNAL_UART=$(USE_EXTERNAL_UART) -DMEM_SIZE=$(MEM_SIZE) -DPLL400=$(PLL400) -DRPSCLK=$(RPSCLK) -DTEST_BRD=$(TEST_BRD) -DUSE_SATA=$(USE_SATA) -DUSE_SATA_ENV=$(USE_SATA_ENV) -DUSE_FLASH=$(USE_FLASH)
+PLATFORM_CPPFLAGS += -DTEXT_BASE=$(TEXT_BASE)
diff --git a/board/oxnas/ide-810.c b/board/oxnas/ide-810.c
index 03d5a0c..ca3b225 100644
--- a/board/oxnas/ide-810.c
+++ b/board/oxnas/ide-810.c
@@ -22,6 +22,8 @@
  */
 #include <common.h>
 
+#if (NAS_VERSION == 810)
+
 #define SATA_DMA_CHANNEL 0
 
 #define DMA_CTRL_STATUS      (0x0)
@@ -889,3 +891,5 @@ int ide_preinit(void)
 
     return !num_disks_found;
 }
+
+#endif /* (NAS_VERSION == 810) */
diff --git a/board/oxnas/platform-800.S b/board/oxnas/platform-800.S
index 8fde117..e5ee37f 100644
--- a/board/oxnas/platform-800.S
+++ b/board/oxnas/platform-800.S
@@ -26,6 +26,8 @@
 #include <config.h>
 #include <version.h>
 
+#if (NAS_VERSION == 800)
+
 /* use estimate of processor speed to calculate number of cycles delay */
 /* delay count is nominal (PLL200 frequency x delay time) / loop count
  * expressing 200us as 200/1000000 and re-arranging gives the expression below
@@ -251,3 +253,5 @@ init_table:
 	.word 0x4580002C, 0x00000000	// Do NOT disable HPROT, ie want write coherency
 
 .ltorg
+
+#endif /* (NAS_VERSION == 800) */
diff --git a/board/oxnas/platform-810.S b/board/oxnas/platform-810.S
index f621f72..b9275d4 100644
--- a/board/oxnas/platform-810.S
+++ b/board/oxnas/platform-810.S
@@ -26,6 +26,8 @@
 #include <config.h>
 #include <version.h>
 
+#if (NAS_VERSION == 810)
+
 /* use estimate of processor speed to calculate number of cycles delay */
 /* delay count is nominal (PLL200 frequency x delay time) / loop count
  * expressing 200us as 200/1000000 and re-arranging gives the expression below
@@ -474,3 +476,5 @@ probe_table:
     .word 0x50000040, 0xabcdef01
 
 .ltorg
+
+#endif /* (NAS_VERSION == 810) */
diff --git a/include/configs/oxnas.h b/include/configs/oxnas.h
index a8ff3cf..966e13f 100644
--- a/include/configs/oxnas.h
+++ b/include/configs/oxnas.h
@@ -29,7 +29,68 @@
 #define writeb(v, p) (*(volatile u8 *)(p)= (v))
 #define writel(v, p) (*(volatile u32*)(p)=(v))
 
-#define CFG_FLASH_EMPTY_INFO
+/*
+ * Defaults (original from board/oxnas/config.mk)
+ */
+#ifndef	NAS_VERSION
+#define	NAS_VERSION	810 /* OXE810 */
+#endif
+
+#ifndef	PLL400
+#define	PLL400		733333333
+#endif
+
+#ifndef	RPSCLK
+#define	RPSCLK		25000000
+#endif
+
+#ifndef	FPGA
+#define	FPGA		0 /* no external FPGA */
+#endif
+
+#ifndef	FPGA_ARM_CLK
+#define	FPGA_ARM_CLK	25000000
+#endif
+
+#ifndef	PROBE_MEM_SIZE
+#define	PROBE_MEM_SIZE	1
+#endif
+
+#ifndef	MEM_SIZE
+#define	MEM_SIZE	64 /* Memory size in MB if probing is not enabled */
+#endif
+
+#ifndef	MEM_ODT
+#define	MEM_ODT		150
+#endif
+
+#ifndef	USE_SATA
+#define	USE_SATA	1
+#endif
+
+#ifndef	USE_SATA_ENV
+#define	USE_SATA_ENV	1
+#endif
+
+#ifndef	USE_FLASH
+#define	USE_FLASH	1 /* use and init CFI/Flash interface */
+#endif
+
+#ifndef	LINUX_ROOT_RAIDED
+#define	LINUX_ROOT_RAIDED 1 /* Linux RootFS on SoftRAID (/dev/md1) */
+#endif
+
+#ifndef	USE_EXTERNAL_UART
+#define	USE_EXTERNAL_UART 0 /* no external UART */
+#endif
+
+#ifndef	INTERNAL_UART
+#define	INTERNAL_UART	2 /* use internal UART2 as console */
+#endif
+
+#ifndef	TEST_BRD
+#define	TEST_BRD	0 /* Only significant for OX800 */
+#endif
 
 /**
  * Architecture
@@ -219,7 +280,6 @@
  * Monitor commands
  */
 #define BASE_COMMANDS (CFG_CMD_IMI    | \
-                       CFG_CMD_IMLS   | \
                        CFG_CMD_BDI    | \
                        CFG_CMD_NET    | \
                        CFG_CMD_PING   | \
@@ -230,7 +290,7 @@
 #ifdef CFG_NO_FLASH
 #define FLASH_COMMANDS (BASE_COMMANDS)
 #else
-#define FLASH_COMMANDS (BASE_COMMANDS | CFG_CMD_FLASH)
+#define FLASH_COMMANDS (BASE_COMMANDS | CFG_CMD_FLASH | CFG_CMD_IMLS)
 #endif // CFG_NO_FLASH
 
 #ifdef CONFIG_OXNAS_USE_SATA
@@ -293,10 +353,10 @@
 #define CFG_FLASH_CFI			1
 #define CFG_FLASH_CFI_DRIVER	1
 
-#define NUM_FLASH_MAIN_BLOCKS   63          /* For Intel 28F320B3T */
-#define NUM_FLASH_PARAM_BLOCKS  8           /* For Intel 28F320B3T */
-#define FLASH_MAIN_BLOCK_SIZE   (64*1024)   /* For Intel 28F320B3T family */
-#define FLASH_PARAM_BLOCK_SIZE  (8*1024)    /* For Intel 28F320B3T family */
+#define NUM_FLASH_MAIN_BLOCKS   63		/* For Intel 28F320B3T */
+#define NUM_FLASH_PARAM_BLOCKS  8		/* For Intel 28F320B3T */
+#define FLASH_MAIN_BLOCK_SIZE   (64*1024)	/* For Intel 28F320B3T family */
+#define FLASH_PARAM_BLOCK_SIZE  (8*1024)	/* For Intel 28F320B3T family */
 
 /* Assuming counts main blocks and parameter blocks, as the Intel/AMD detection */
 /* I'm intending to copy would seem to indicate */
@@ -311,7 +371,9 @@
 #define CFG_FLASH_WRITE_TOUT    (20*CFG_HZ)	/* Timeout for Flash Write */
 #define CFG_FLASH_WRITE_ATTEMPTS 5
 
-#define STATIC_BUS_FLASH_CONFIG 0x4f1f3f3f  /* Slow ASIC settings */
+#define STATIC_BUS_FLASH_CONFIG 0x4f1f3f3f	/* Slow ASIC settings */
+
+#define CFG_FLASH_EMPTY_INFO		/* print 'E' for empty sector on flinfo */
 
 #endif // !CFG_NO_FLASH
 
diff --git a/mkconfig b/mkconfig
old mode 100644
new mode 100755
-- 
1.7.1

